scripts/ec2/tags.coffee

```diff
@@ -1,26 +1,47 @@
-
-cson = require 'cson'
 ec2 = require('../../ec2.coffee')
-util = require 'util'
+moment = require('moment')
+
 
-# Expose to other scripts
 tags =
 
-  addSchedule: (msg, instances, schedule="8:18") ->
-    ec2.createTags {Resources: instances, Tags: [Key: "Schedule", Value: schedule]}, (err, res) ->
-      if err
-        console.log "Error creating tags: #{err}"
-        if msg
-          return msg.send "Error creating tags: #{err}" if err
+  DEFAULT_SCHEDULE: "8:18"
+
+  SCHEDULE_TAG: "RunSchedule"
+
+  # convert our concise schedule strings ("8:18") to the new UTC-based format
+  formatSchedule: (hours) ->
+    [_, start, stop] = hours.match /^(\d{1,2}):(\d{1,2})$/
+    startUtc = moment({hour: start, minute: 0}).utc().hour().toString()
+    stopUtc = moment({hour: stop, minute: 0}).utc().hour().toString()
+    [
+      "#{startUtc.padStart(2, '0').padEnd(4, '0')}",
+      "#{stopUtc.padStart(2, '0').padEnd(4, '0')}",
+      "utc",
+      "mon,tue,wed,thu,fri",
+    ].join ";"
+
+  addSchedule: (msg, instances, schedule = tags.DEFAULT_SCHEDULE) ->
+    try
```

Comment:
From above, if the code catches parser errors here, then I'm wondering if it's necessary to do such explicit matching above.

Reply:
@champain No, I think you're right - we can do more validation of the inputs. This won't catch something like `99:88` – we'll get some `NaN`s in that case. I'll improve this, thanks for the feedback.

Reply:
👍